hashicorp/dev-portal
# feat: add version alert to api-docs

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-leah-featold-api-docs-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1207339219333499/1205587495423439/f) 🎟️

## 🗒️ What

This PR adds an alert banner when viewing an older or preview version of the API docs.

## 🤷 Why

So it is clear to the user that the api docs may not align with the current stable version.

## 🛠️ How

I added the `ApiDocsVersionAlert` to `OpenApiDocsView`. This alert was already created for other API docs on the dev site. I also adjusted the alert to use a dynamic link and passed the correct props through to the `OpenApiDocsView` so the alert will render as expected.

## 📸 Design Screenshots



Preview
<img width="2464" alt="Screenshot 2024-10-22 at 4 14 09 PM" src="https://github.com/user-attachments/assets/7edcd930-1d94-4032-95b9-86f62be44d9c">
<img width="2459" alt="Screenshot 2024-10-22 at 4 13 49 PM" src="https://github.com/user-attachments/assets/c2d7f74d-d857-43bb-be5c-3a447641ef5d">


Older version
<img width="2463" alt="Screenshot 2024-10-22 at 4 12 41 PM" src="https://github.com/user-attachments/assets/25761a37-881a-4cc5-9a7e-abe79ce963f6">
<img width="2462" alt="Screenshot 2024-10-22 at 4 13 01 PM" src="https://github.com/user-attachments/assets/7b2811af-8860-473f-8778-a2ecf69b100d">


## 🧪 Testing



Preview:
1. Go to [packer api docs link](https://dev-portal-git-leah-featold-api-docs-hashicorp.vercel.app/hcp/api-docs/vault-secrets/2023-11-28)
2. Notice that there is an alert banner at the top that signifies that you are on a preview docs version
3. Click on the link in the banner
4. Verify that it brings you to the stable docs version for the correct docs

Older version:
1. Go to [vault secrets api docs link](https://dev-portal-git-leah-featold-api-docs-hashicorp.vercel.app/hcp/api-docs/packer/2021-04-30)
2. Notice that there is an alert banner at the top that signifies that you are on a different docs version
3. Click on the link in the banner
4. Verify that it brings you to the stable docs version for the correct docs
